explorer/src/utils/name-service.tsx

```diff
@@ -55,6 +55,30 @@ export async function findOwnedNameAccountsForUser(
   return accounts.map((a) => a.publicKey);
 }
 
+async function retrieveDomainInfo(
+  name: PublicKey,
+  centralState: PublicKey,
+  connection: Connection
+) {
+  const domainKey = await getDomainKey(name.toBase58(), centralState);
+  let nameRegistry: NameRegistryState | undefined;
+  try {
+    const registry = await NameRegistryState.retrieve(connection, domainKey);
+    nameRegistry = registry.registry;
+  } catch {}
+
+  if (!nameRegistry?.data) {
+    return;
+  }
+
+  const nameLength = new BN(nameRegistry!.data.slice(0, 4), "le").toNumber();
+  return {
+    name: nameRegistry.data.slice(4, 4 + nameLength).toString() + ".sol",
+    address: domainKey,
+    class: nameRegistry.class,
+  };
+}
+
 export async function performReverseLookup(
```

Comment:
Can you migrate to using the `performReverseLookup` function provided by the name service package? https://github.com/Bonfida/solana-program-library/blob/dc0b43d690087d1fb12873d1f3c8218bee8e368a/name-service/js/src/utils.ts#L170